src/azure-cli/azure/cli/command_modules/appconfig/_snapshotmodels.py

```diff
@@ -0,0 +1,222 @@
+# --------------------------------------------------------------------------------------------
+
+from datetime import datetime
+from enum import Enum
+
+# pylint: disable=too-few-public-methods
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=line-too-long
+
+
+class SnapshotQueryFields(Enum):
+    NAME = 0x0001
+    STATUS = 0x0002
+    FILTERS = 0x0004
+    COMPOSITION_TYPE = 0x0008
+    CREATED = 0x0010
+    EXPIRES = 0x0020
+    SIZE = 0x0040
+    ITEMS_COUNT = 0x0080
+    TAGS = 0x0100
+    STATUS_CODE = 0x0200
+    RETENTION_PERIOD = 0x0400
+    ETAG = 0x0800
+    ALL = NAME | STATUS | FILTERS | COMPOSITION_TYPE | CREATED | EXPIRES | SIZE | ITEMS_COUNT | TAGS | RETENTION_PERIOD | ETAG | STATUS_CODE
+
+
+class Snapshot:
+    '''
+    Class used to represent the Snapshot entity returned from requests
+
+    :ivar str name:
+        Name of the snapshot.
+    :ivar str status:
+        Provisioning status of the snapshot.
+    :ivar dict[str, str] filters:
+        Dictionary of filters used in building snapshot.
+    :ivar str etag:
+        The ETag contains a value that you can use to perform operations.
+    :ivar str composition_type:
+        Choose how snapshot filters are applied in building snapshot.
+    :ivar str created:
+        Time of creation of snapshot.
+    :ivar str expires:
+        The time at which an archived snapshot will be deleted.
+    :ivar int size:
+        The size in bytes of the built snapshot.
+    :ivar int items_count:
+        The number of key-values in the built snapshot.
+    :ivar dict[str, str] tags:
+        Dictionary of tags of the snapshot.
+    :ivar int retention_period:
+        Number of seconds for which an archived snapshot will be kept before being deleted.
+    :ivar int status_code:
+        The status code returned during the snapshot creation process.
+    '''
+
+    def __init__(self,
+                 name,
+                 status,
+                 filters,
+                 etag=None,
+                 status_code=None,
+                 composition_type=None,
+                 created=None,
+                 expires=None,
+                 size=None,
+                 items_count=None,
+                 tags=None,
+                 retention_period=None,
+                 ):
+
+        self.name = name
+        self.status = status
+        self.filters = filters
+        self.status_code = status_code
+        self.etag = etag
+        self.composition_type = composition_type
+        self.created = created.isoformat() if isinstance(created, datetime) else str(created)
+        self.expires = expires.isoformat() if isinstance(expires, datetime) else (str(expires) if expires else None)
+        self.size = size
+        self.items_count = items_count
+        self.tags = tags
+        self.retention_period = retention_period
+
+    def __str__(self):
+        return "\netag: " + self.etag + \
```

Comment:
If other fields are starting with capital letter, I think etag should also start with capital E.

Reply:
This has been updated, thanks!